src/main/java/com/mentoring/ecommerce/adapter/out/persistence/postgresql/product/ProductPostgreSQLAdapter.java

```diff
@@ -0,0 +1,35 @@
+package com.mentoring.ecommerce.adapter.out.persistence.postgresql.product;
+
+import com.mentoring.common.exceptions.ProductNotFoundException;
+import com.mentoring.ecommerce.application.port.out.FindProductPort;
+import com.mentoring.ecommerce.application.port.out.SaveProductPort;
+import com.mentoring.ecommerce.domain.Product;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Component
+public class ProductPostgreSQLAdapter implements FindProductPort, SaveProductPort {
+
+    ProductPostgreSQLRepository repository;
+
+    @Override
+    public List<Product> findAll() {
+        return (List<Product>) repository.findAll();
+    }
+
+    @Override
+    public Product findById(Integer id) {
+        Optional<Product> product = repository.findById(id);
+        product.orElseThrow(() -> new ProductNotFoundException("Product not found: " + id));
```

Comment:
To avoid multiple validations every time `findById` is called, I put it in the `persistence` layer and remove it from the `Service` layer _(line 18)_. That way you will only have one `not found by id` validation and not in all `Services` that call this method.

Can I keep it here or should I revert to the way it was?

Reply:
I understand that this makes the code have less lines, however, it is not the responsibility of the persistence layer to throw exceptions. It's purpose is to: retrieve the Product by id. If it is not found, it will should simply return null.

Reply:
Exception moved to the Service layer in the M-05 MR.